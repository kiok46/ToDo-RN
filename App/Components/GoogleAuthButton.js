import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  ActivityIndicator,
  View,  
} from 'react-native';
import PropTypes from 'prop-types';

import Colors from '../Utils/Colors';


class GoogleAuthButton extends Component {

  render() {
      if (this.props.showActivity) {
          return (
            <ActivityIndicator size="large"/>
          )
      } else {
        return (
            <TouchableOpacity onPress={() => this.props.onPress()}>
                <View style={[styles.authButtonStyle, {backgroundColor: Colors.googleAuthColor}]}>
                    <Text style={{ color: Colors.white }}> Login with Google </Text>
                </View>
            </TouchableOpacity>
        );
      }
  }
}

const styles = StyleSheet.create({
  authButtonStyle: {
      borderRadius: 5,
      padding: 10
  }
});

GoogleAuthButton.defaultProps = {
    onPress: () => {},
    showActivity: false
}

GoogleAuthButton.propTypes = {
    onPress: PropTypes.func,
    showActivity: PropTypes.bool
}


export default GoogleAuthButton;

