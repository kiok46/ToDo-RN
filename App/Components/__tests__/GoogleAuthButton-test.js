import React from 'react';
import GoogleAuthButton from '../GoogleAuthButton';
import { shallow } from 'enzyme';


describe('testing GoogleAuthButton rendering', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <GoogleAuthButton />
        )
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ showActivity: true })
        expect(wrapper).toMatchSnapshot();
    })
})
