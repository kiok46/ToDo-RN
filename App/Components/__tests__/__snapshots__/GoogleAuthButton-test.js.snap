// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GoogleAuthButton rendering renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleAuthButton
    onPress={[Function]}
    showActivity={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Array [
            Object {
              "borderRadius": 5,
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(221, 75, 57, 1)",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
            }
          }
        >
           Login with Google 
        </Text>
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
            }
          }
        >
           Login with Google 
        </Text>,
        "style": Array [
          Object {
            "borderRadius": 5,
            "padding": 10,
          },
          Object {
            "backgroundColor": "rgba(221, 75, 57, 1)",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " Login with Google ",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "rgba(255, 255, 255, 1)",
          },
        },
        "ref": null,
        "rendered": " Login with Google ",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Array [
              Object {
                "borderRadius": 5,
                "padding": 10,
              },
              Object {
                "backgroundColor": "rgba(221, 75, 57, 1)",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
              }
            }
          >
             Login with Google 
          </Text>
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
              }
            }
          >
             Login with Google 
          </Text>,
          "style": Array [
            Object {
              "borderRadius": 5,
              "padding": 10,
            },
            Object {
              "backgroundColor": "rgba(221, 75, 57, 1)",
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " Login with Google ",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "rgba(255, 255, 255, 1)",
            },
          },
          "ref": null,
          "rendered": " Login with Google ",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`testing GoogleAuthButton rendering renders as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleAuthButton
    onPress={[Function]}
    showActivity={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "color": "#999999",
      "hidesWhenStopped": true,
      "size": "large",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "color": "#999999",
        "hidesWhenStopped": true,
        "size": "large",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
